### PR TITLE
[TEST][CUDNN] xfail SDPA on pre-Ampere

### PR DESCRIPTION
**Pull request by Aidyn-A**
All tests in `test_cudnn_executor.py` fail on Volta (and older) GPUs with the following message:
```
RuntimeError: No valid engine configs for Matmul_MUL_GEN_INDEX_GEN_INDEX_CMP_GE_BINARY_SELECT_Reduction_SUB_EXP_Reduction_LOG_ADD_DIV_Matmul_
```
This PR simply adds the `xfails` as CUDNN SDPA requires Ampere+.

### CODE REVIEW

File: thunder/tests/test_cudnn_executor.py [hunk 1]
```diff
@@ -21,6 +21,15 @@
 from thunder.executors.cudnnex import cudnn_ex, cudnn_version
 
 
+def _maybe_xfail() -> None:
+    dev: torch.device = thunder.core.devices.to_torch_device("cuda:0")
+    cuda_major: int
+    cuda_minor: int
+    cuda_major, cuda_minor = torch.cuda.get_device_capability(dev)
+    if cuda_major < 8:
+        pytest.xfail("cuDNN SDPA uses flash attention, which requires Ampere+")
```
Comment (crcrpar) [inline thunder/tests/test_cudnn_executor.py:30]: nit-picking

```suggestion
    cuda_major, _ = torch.cuda.get_device_capability(dev)
    if cuda_major < 8:
        pytest.xfail("cuDNN SDPA uses flash attention, which requires Ampere+")
```

**Review by crcrpar — APPROVED** [review]

**Review by vedaanta-nvidia — APPROVED** [review]
Probably cleaner to do it in a decorator? https://docs.pytest.org/en/stable/how-to/skipping.html#xfail-mark-test-functions-as-expected-to-fail

Looks fine as is too though.

**Comment by Aidyn-A** [comment]
cc @mruberry

**Review by carmocca — APPROVED** [review]
Thanks!

**Pull request merged**